src/rsz/src/RepairDesign.cc

```diff
@@ -2019,6 +2051,27 @@ void RepairDesign::makeRepeater(
     sta_->connectPin(buffer, buffer_output_port, buffer_op_net);
   }  // case 2
 
+  if (!connections_modified) {
```

Comment:
When would this happen?  It seems both cases modify connections.

Reply:
You can still insert a repeater in that case unless the driver is also dont-touch or a primary input

Reply:
you cannot, because you cannot change the net that the dont_touch buffers are attached to (thats what is failing). If you cannot break the net into two parts because all the instances on one or the other side of it are marked dont_touch then adding the buffer just results in a floating net somewhere.

Reply:
You just leave the loads on their current net and add a new net at the repeater input instead.

Reply:
In any case it is better to avoid this upstream impossible combinations

Reply:
I mode the check up to earlier, so the original code is not really modified.